src/electron/process_manager.ts

```diff
@@ -85,10 +87,21 @@ function testTapDevice() {
   }
 }
 
-async function isSsLocalReachable() {
-  await isServerReachable(
-      PROXY_ADDRESS, PROXY_PORT, SSLOCAL_CONNECTION_TIMEOUT, SSLOCAL_MAX_ATTEMPTS,
-      SSLOCAL_RETRY_INTERVAL_MS);
+interface Process {
+  start(): Promise<void>;
+  stop(): void;
+  onExit?: ((code: number) => void);
+  enableDebugMode(): void;
+}
+
+interface ShadowsocksProcess extends Process {
+  // Checks whether the network and proxy support UDP forwarding and validates
+  // that the proxy credentials are valid. Throws an error if any of the checks fail.
+  checkConnectivity(): Promise<void>;
+}
+
+interface Tun2socksProcess extends Process {
+  isUdpEnabled: boolean;
```

Comment:
Interfaces with fields tend to be a red flag (with exceptions like serialization formats)
Why don't you pass this to start() instead?